libraries/AP_Proximity/AP_Proximity_MAV.cpp

```diff
@@ -164,12 +164,67 @@ void AP_Proximity_MAV::handle_msg(const mavlink_message_t &msg)
             }
         }
 
-        // update proximity sectors validity and boundary point
-        for (uint8_t i = 0; i < PROXIMITY_NUM_SECTORS; i++) {
-            _distance_valid[i] = (_distance[i] >= _distance_min) && (_distance[i] <= _distance_max);
-            if (sector_updated[i]) {
-                update_boundary_for_sector(i, false);
-            }
+        // process the last face
+        if (face_distance_valid) {
+            boundary.set_face_attributes(face, face_yaw_deg, face_distance);
+        } else {
+            boundary.reset_face(face);
+        }
+        return;
+    }
+
+    if (msg.msgid == MAVLINK_MSG_ID_OBSTACLE_DISTANCE_3D) {
+        mavlink_obstacle_distance_3d_t packet;
+        mavlink_msg_obstacle_distance_3d_decode(&msg, &packet);
```

Comment:
Please break out a `handle_obstacle_distance_3d(const mavlink_message_t &msg)` method.

No need to continue this mega-method pattern we've previously had in the mavlink routines.

Reply:
You also need to check the message's frame and reject the frames other than what you expect.

Reply:
I have broken down the entire driver into three methods for each mavlink message type